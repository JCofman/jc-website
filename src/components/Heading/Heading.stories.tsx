import React from 'react';
import { storiesOf } from '@storybook/react';
import Heading, { APPEARANCES } from './Heading';

storiesOf(`Heading`, module).add(`Basic`, () => (
  <>
    <Heading appearance={APPEARANCES.H1}> Heading H1 </Heading>
    <Heading appearance={APPEARANCES.H2}> Heading H2 </Heading>
    <Heading appearance={APPEARANCES.H3}> Heading H3 </Heading>
    <Heading appearance={APPEARANCES.H4}> Heading H4 </Heading>
    <Heading appearance={APPEARANCES.H5}> Heading H4 </Heading>
    <Heading appearance={APPEARANCES.H6}> Heading H4 </Heading>
  </>
));
