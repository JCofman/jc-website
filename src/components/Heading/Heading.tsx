import React from 'react';
import PropTypes from 'prop-types';
import {
  StyledHeadingH1,
  StyledHeadingH2,
  StyledHeadingH3,
  StyledHeadingH4,
  StyledHeadingH5,
  StyledHeadingH6,
} from './StyledHeading';

export const APPEARANCES = {
  H1: `H1`,
  H2: `H2`,
  H3: `H3`,
  H4: `H4`,
  H5: `H5`,
  H6: `H6`,
};

export const BACKGROUND_STYLES = {
  PARALLELOGRAM: 'PARALLELOGRAM',
  TRIANGLE: 'TRIANGLE',
  CIRCLE: 'CIRCLE',
};

const Heading = ({ appearance, children, backgroundStyle = 'TRIANGLE', ...rest }) => {
  const { ...headingProps } = rest;
  switch (appearance) {
    case `H1`:
      return (
        <StyledHeadingH1 backgroundStyle={backgroundStyle} {...headingProps}>
          {children}
        </StyledHeadingH1>
      );
    case `H2`:
      return (
        <StyledHeadingH2 backgroundStyle={backgroundStyle} {...headingProps}>
          {children}
        </StyledHeadingH2>
      );
    case `H3`:
      return (
        <StyledHeadingH3 backgroundStyle={backgroundStyle} {...headingProps}>
          {children}
        </StyledHeadingH3>
      );
    case `H4`:
      return (
        <StyledHeadingH4 backgroundStyle={backgroundStyle} {...headingProps}>
          {children}
        </StyledHeadingH4>
      );
    case `H5`:
      return (
        <StyledHeadingH5 backgroundStyle {...headingProps}>
          {children}
        </StyledHeadingH5>
      );
    case `H6`:
      return (
        <StyledHeadingH6 backgroundStyle {...headingProps}>
          {children}
        </StyledHeadingH6>
      );
    default:
      return null;
  }
};

Heading.displayName = `Heading`;

Heading.propTypes = {
  /** any nodes to be rendered (usually text nodes) */
  children: PropTypes.any,

  /** typography of the heading */
  appearance: PropTypes.oneOf(Object.keys(APPEARANCES)),
};

Heading.defaultProps = {
  appearance: APPEARANCES.H1,
};

export default Heading;
