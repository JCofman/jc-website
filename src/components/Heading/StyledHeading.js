import styled, { css } from 'styled-components';

const sharedHeadingStyles = css`
  font-family: 'Raleway', sans-serif;
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
  a {
    color: var(--color-text);
    transition: color var(--theme-transition);
  }
  a:hover,
  a:focus {
    color: var(--color-primary);
    transition: none;
  }
  ${(props) =>
    props.backgroundStyle === 'TRIANGLE' &&
    `:before {
    width: 0;
    height: 0;
    opacity: 0.2;
    border-left: 1em solid transparent;
    border-right: 1em solid transparent;
    border-bottom: 2em solid var(--color-text);
    content: '';
    pointer-events: none;
    position: absolute;
    z-index: -1;
    transform: translateX(-0.5em) translateY(-1.5rem);
  }`}
  ${(props) =>
    props.backgroundStyle === 'PARALLELOGRAM' &&
    `:before {
    width: 50px;
    height: 40px;
    background: var(--color-text);
    opacity: 0.2;
    content: '';
    pointer-events: none;
    position: absolute;
    transform: translateX(-0.5em) translateY(-1.5rem) skew(20deg);
  }`}
  ${(props) =>
    props.backgroundStyle === 'CIRCLE' &&
    `:before {
    width: 50px;
    height: 50px;
    border-radius: 50%;
    opacity: 0.2;
    background-color: var(--color-text);
    content: '';
    pointer-events: none;
    position: absolute;
    transform: translateX(-0.5em) translateY(-1.5rem);
  }`}
`;

const StyledHeadingH1 = styled.h1`
  font-size: var(--font-size-7xl);
  ${sharedHeadingStyles}
`;

const StyledHeadingH2 = styled.h2`
  ${sharedHeadingStyles}
  font-size: var(--font-size-5xl);
`;

const StyledHeadingH3 = styled.h3`
  ${sharedHeadingStyles}
  font-size: var(--font-size-4xl);
`;
const StyledHeadingH4 = styled.h4`
  ${sharedHeadingStyles}
  font-size: var(--font-size-3xl);
`;
const StyledHeadingH5 = styled.h5`
  ${sharedHeadingStyles}
  font-size: var(--font-size-2xl);
`;
const StyledHeadingH6 = styled.h6`
  ${sharedHeadingStyles}
  font-size: var(--font-size-xl);
`;

export { StyledHeadingH1, StyledHeadingH2, StyledHeadingH3, StyledHeadingH4, StyledHeadingH5, StyledHeadingH6 };
