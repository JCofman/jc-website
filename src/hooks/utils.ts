const throttle = (func, threshold = 250, scope) => {
  let last, deferTimer;
  return function () {
    let context = scope || this;

    let now = Date.now(),
      args = arguments;
    if (last && now < last + threshold) {
      // hold on to it
      clearTimeout(deferTimer);
      deferTimer = setTimeout(function () {
        last = now;
        func.apply(context, args);
      }, threshold);
    } else {
      last = now;
      func.apply(context, args);
    }
  };
};

export { throttle };
