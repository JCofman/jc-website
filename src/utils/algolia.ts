// const postQuery = `{
//     posts: allMarkdownRemark(
//       filter: { fileAbsolutePath: { regex: "/posts/" } }
//     ) {
//       edges {
//         node {
//           objectID: id
//           frontmatter {
//             title
//             path
//             date(formatString: "MMM D, YYYY")
//             tags
//           }
//           excerpt(pruneLength: 5000)
//         }
//       }
//     }
//   }`;
// const flatten = arr =>
//   arr.map(({ node: { frontmatter, ...rest } }) => ({
//     ...frontmatter,
//     ...rest,
//   }));
// const settings = { attributesToSnippet: [`excerpt:20`] };
// const queries = [
//   {
//     query: pageQuery,
//     transformer: ({ data }) => flatten(data.pages.edges),
//     indexName: `Pages`,
//     settings,
//   },
//   {
//     query: postQuery,
//     transformer: ({ data }) => flatten(data.posts.edges),
//     indexName: `Posts`,
//     settings,
//   },
// ];
// module.exports = queries;
