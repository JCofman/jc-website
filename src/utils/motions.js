const iconMotion = {
  rest: {
    rotate: 0,
  },
  hover: {
    rotate: 9,
    transition: { type: 'spring', stiffness: 900 },
  },
};

export { iconMotion };
