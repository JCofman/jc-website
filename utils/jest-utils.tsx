import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { themes, breakPoints, maxWidth, themeTransition, colors } from '../src/components/Layout/Theme';

export const renderWithTheme = (component) =>
  render(
    <ThemeProvider
      theme={{
        mode: themes[`light`],
        ...breakPoints,
        maxWidth,
        themeTransition,
        colors: { ...colors },
      }}
    >
      {component}
    </ThemeProvider>
  );
