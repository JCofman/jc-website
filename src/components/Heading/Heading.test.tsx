import React from 'react';

import { renderWithTheme } from '../../../utils/jest-utils';
import Heading, { APPEARANCES } from './Heading';

Object.keys(APPEARANCES).forEach(appearance => {
  test(`should render a ${appearance} header tags`, () => {
    const { container } = renderWithTheme(<Heading appearance={appearance}>I am a {appearance} headline</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
