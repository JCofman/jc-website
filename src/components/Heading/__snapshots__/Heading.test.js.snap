// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a H1 header tags 1`] = `
.c0 {
  font-size: var(--font-size-7xl);
  font-family: 'Raleway',sans-serif;
  -webkit-transform: skew(-5deg) rotate(-1deg);
  -ms-transform: skew(-5deg) rotate(-1deg);
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
}

.c0 a {
  color: var(--color-text);
  -webkit-transition: color var(--theme-transition);
  transition: color var(--theme-transition);
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-primary);
  -webkit-transition: none;
  transition: none;
}

.c0:before {
  width: 0;
  height: 0;
  opacity: 0.2;
  border-left: 1em solid transparent;
  border-right: 1em solid transparent;
  border-bottom: 2em solid var(--color-text);
  content: '';
  pointer-events: none;
  position: absolute;
  z-index: -1;
  -webkit-transform: translateX(-0.5em) translateY(-1.5rem);
  -ms-transform: translateX(-0.5em) translateY(-1.5rem);
  transform: translateX(-0.5em) translateY(-1.5rem);
}

<h1
  class="c0"
>
  I am a 
  H1
   headline
</h1>
`;

exports[`should render a H2 header tags 1`] = `
.c0 {
  font-family: 'Raleway',sans-serif;
  -webkit-transform: skew(-5deg) rotate(-1deg);
  -ms-transform: skew(-5deg) rotate(-1deg);
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
  font-size: var(--font-size-5xl);
}

.c0 a {
  color: var(--color-text);
  -webkit-transition: color var(--theme-transition);
  transition: color var(--theme-transition);
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-primary);
  -webkit-transition: none;
  transition: none;
}

.c0:before {
  width: 0;
  height: 0;
  opacity: 0.2;
  border-left: 1em solid transparent;
  border-right: 1em solid transparent;
  border-bottom: 2em solid var(--color-text);
  content: '';
  pointer-events: none;
  position: absolute;
  z-index: -1;
  -webkit-transform: translateX(-0.5em) translateY(-1.5rem);
  -ms-transform: translateX(-0.5em) translateY(-1.5rem);
  transform: translateX(-0.5em) translateY(-1.5rem);
}

<h2
  class="c0"
>
  I am a 
  H2
   headline
</h2>
`;

exports[`should render a H3 header tags 1`] = `
.c0 {
  font-family: 'Raleway',sans-serif;
  -webkit-transform: skew(-5deg) rotate(-1deg);
  -ms-transform: skew(-5deg) rotate(-1deg);
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
  font-size: var(--font-size-4xl);
}

.c0 a {
  color: var(--color-text);
  -webkit-transition: color var(--theme-transition);
  transition: color var(--theme-transition);
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-primary);
  -webkit-transition: none;
  transition: none;
}

.c0:before {
  width: 0;
  height: 0;
  opacity: 0.2;
  border-left: 1em solid transparent;
  border-right: 1em solid transparent;
  border-bottom: 2em solid var(--color-text);
  content: '';
  pointer-events: none;
  position: absolute;
  z-index: -1;
  -webkit-transform: translateX(-0.5em) translateY(-1.5rem);
  -ms-transform: translateX(-0.5em) translateY(-1.5rem);
  transform: translateX(-0.5em) translateY(-1.5rem);
}

<h3
  class="c0"
>
  I am a 
  H3
   headline
</h3>
`;

exports[`should render a H4 header tags 1`] = `
.c0 {
  font-family: 'Raleway',sans-serif;
  -webkit-transform: skew(-5deg) rotate(-1deg);
  -ms-transform: skew(-5deg) rotate(-1deg);
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
  font-size: var(--font-size-3xl);
}

.c0 a {
  color: var(--color-text);
  -webkit-transition: color var(--theme-transition);
  transition: color var(--theme-transition);
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-primary);
  -webkit-transition: none;
  transition: none;
}

.c0:before {
  width: 0;
  height: 0;
  opacity: 0.2;
  border-left: 1em solid transparent;
  border-right: 1em solid transparent;
  border-bottom: 2em solid var(--color-text);
  content: '';
  pointer-events: none;
  position: absolute;
  z-index: -1;
  -webkit-transform: translateX(-0.5em) translateY(-1.5rem);
  -ms-transform: translateX(-0.5em) translateY(-1.5rem);
  transform: translateX(-0.5em) translateY(-1.5rem);
}

<h4
  class="c0"
>
  I am a 
  H4
   headline
</h4>
`;

exports[`should render a H5 header tags 1`] = `
.c0 {
  font-family: 'Raleway',sans-serif;
  -webkit-transform: skew(-5deg) rotate(-1deg);
  -ms-transform: skew(-5deg) rotate(-1deg);
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
  font-size: var(--font-size-2xl);
}

.c0 a {
  color: var(--color-text);
  -webkit-transition: color var(--theme-transition);
  transition: color var(--theme-transition);
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-primary);
  -webkit-transition: none;
  transition: none;
}

<h5
  class="c0"
>
  I am a 
  H5
   headline
</h5>
`;

exports[`should render a H6 header tags 1`] = `
.c0 {
  font-family: 'Raleway',sans-serif;
  -webkit-transform: skew(-5deg) rotate(-1deg);
  -ms-transform: skew(-5deg) rotate(-1deg);
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: var(--color-text);
  text-shadow: var(--text-shadow);
  font-size: var(--font-size-xl);
}

.c0 a {
  color: var(--color-text);
  -webkit-transition: color var(--theme-transition);
  transition: color var(--theme-transition);
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-primary);
  -webkit-transition: none;
  transition: none;
}

<h6
  class="c0"
>
  I am a 
  H6
   headline
</h6>
`;
